### PR TITLE
Typo and spelling check XML doc and strings. A to E. #Hacktoberfest

### PR DESCRIPTION
**Pull request by bchavez**
:dog2: *Much error.* :dog2: 
:dog: *Very typo.* :dog: 
:heavy_check_mark: *Many spell check.* :heavy_check_mark: 

I had to break this up into multiple PRs. This PR only contains **A to E** corrections. Mostly focused on XML doc comments. I tried to ignore `//` source comments. Also, corrected some hard-coded strings in `Exception` messages that had spelling errors.

Let me know if you'd like me to continue checking.

Also, happy #Hacktoberfest ! :smiley_cat: :fearful: :skull_and_crossbones: :moon: :bat: 

Thanks,
Brian

:crystal_ball: :dolls: ***["Magic and technology. Voodoo dolls and chants. Electricity. Were makin weird science..."](https://www.youtube.com/watch?v=Jm-upHSP9KU)***

### CODE REVIEW

File: src/Orleans.Core/Async/AsyncExecutorWithRetries.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@
 namespace Orleans
 {
     /// <summary>
-    /// This class a convinent utiliity class to execute a certain asyncronous function with retires,
+    /// This class a convenient utility class to execute a certain asynchronous function with retires,
```
Comment (ReubenBond) [inline src/Orleans.Core/Async/AsyncExecutorWithRetries.cs]: "is a convenient utility" (missing is)

Comment (philrowan-mtb) [inline src/Orleans.Core/Async/AsyncExecutorWithRetries.cs]: nit: retries is also spelled incorrectly.

Comment (bchavez) [reply src/Orleans.Core/Async/AsyncExecutorWithRetries.cs]: Thank ya' @philrowan-mtb ! :+1:

**Comment by sergeybykov** [comment]
Thank you, @bchavez!

**Comment by bchavez** [comment]
Would you like me to keep going?

I don't want to bother ya'll with spelling/typo PRs if you don't want them.

**Comment by sergeybykov** [comment]
Please, by all means! These are embarrassing typos and misspelling. So happy to see them getting fixed! Thank you, Brian!

**Pull request merged**